develop/codingguidelines/Protocol/Data_handling.md

```diff
@@ -6,41 +6,108 @@ uid: Data_handling
+
+However, there are some cases where you may want a different approach:
+
+- When dealing with too much data to hold in memory, it can become more important to look at memory usage instead of CPU cycles.
+- When dealing with too much data to retrieve in a single call.
+
+In those cases, you may want to consider using a loop and performing only a few GetRow calls instead of a single GetColumns call to retrieve the whole table.
+
+> [!NOTE]
+> In this case, you should always add a comment in your QAction code that explains why you are using a non-standard way of handling table data.
+
+For example:
+
+```txt
+System Information
+------------------
+      Time of this report: 11/14/2019, 12:29:49
+             Machine name: 
+               Machine Id: {7715B613-530C-4E8F-B12D-945A5A54BDC5}
+         Operating System: Windows 10 Pro 64-bit (10.0, Build 17763) (17763.rs5_release.180914-1434)
+                 Language: English (Regional Setting: English)
+      System Manufacturer: Dell Inc.
+             System Model: OptiPlex 3020
+                     BIOS: BIOS Date: 02/15/17 05:19:18 Ver: A15.00  (type: BIOS)
+                Processor: Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz (4 CPUs), ~3.3GHz
+                   Memory: 12288MB RAM
+      Available OS Memory: 12206MB RAM
+                Page File: 11848MB used, 7269MB available
+              Windows Dir: C:\WINDOWS
+          DirectX Version: DirectX 12
+      DX Setup Parameters: Not found
+         User DPI Setting: 96 DPI (100 percent)
+       System DPI Setting: 96 DPI (100 percent)
+          DWM DPI Scaling: Disabled
+                 Miracast: Available, with HDCP
+Microsoft Graphics Hybrid: Not Supported
+           DxDiag Version: 10.00.17763.0001 64bit Unicode
+```
+
+Metrics for a table with 5 columns (average results from looping 100 times with the setup displayed above):
+
+| Number of Rows | GetRow in loop | GetColumns | NT_Notify (external get table in one call) |
+|----------------|----------------|------------|--------------------------------------------|
+| 20             | 44 ms          | 16 ms      | 100 ms                                     |
+| 100            | 260 ms         | 40 ms      | 390 ms                                     |
+| 10 000         | 25 000 ms      | 3000 ms    | 40 500 ms                                  |
+
+As you can see, it might be a lot more efficient in a table with a lot of data to first do a GetColumn of one or two columns and then perform several GetRows in a loop.
+
+Based on experience, we recommend that you follow these basic guidelines:
+
+- Performing a GetColumns:
+
+  - Try to stay below 120 000 cells in 1 get, except if the content of a cell is a 20 MB string or something equally large, in which case you should split up more.
+  - Usually starts breaking down if you go above 100 000 rows (considering an average of 7 columns).
```

Comment:
This "starts breaking down" isn't entirely clear. Could you clarify this?

Reply:
Memory Leaks and eventual process crashes might occur if you go above 100 000 rows (considering an average of 8 columns).